fluent-syntax/src/visitor.js

```diff
@@ -0,0 +1,60 @@
+import { BaseNode } from "./ast";
+
+/*
+ * Abstract Visitor pattern
+ */
+export class Visitor {
+  visit(node) {
+    if (Array.isArray(node)) {
+      node.forEach(child => this.visit(child));
+      return;
+    }
+    if (!(node instanceof BaseNode)) {
+      return;
+    }
+    const nodename = node.type;
+    const visit = this[`visit_${nodename}`] || this.generic_visit;
```

Comment:
Nit: I'd just interpolate `node.type` here, without the interim `nodename` variable. It reads better, too.